src/v/cloud_storage/BUILD

```diff
@@ -13,3 +13,170 @@ redpanda_cc_library(
         "//src/v/serde:uuid",
     ],
 )
+
+redpanda_cc_library(
```

Comment:
I think this is the biggest cc_library I've seen 🏅

Reply:
hah yeh. not sure how much of it is used since the deps are almost entirely driven by missing headers. i guess clang18 can help out here some, or we can do a iwyu pass